ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml

```diff
@@ -87,6 +87,9 @@ tests:
   skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
   steps:
     cluster_profile: aws
+    env:
+      WINDOWS_AWS_AMI_FILTER: Windows_Server-2019-English-Full-ContainersLatest-????.??.??
+      WINDOWS_CONTAINER_IMAGE: mcr.microsoft.com/powershell:lts-nanoserver-1809
```

Comment:
IMO this is embedding too much logic within the release repo. We shouldn't need to open a release PR if we want to change the container image being used. I'd suggest sticking with a single env var which if set indicates 2019 should be used.

Reply:
Everything is about context and in this context, we need a specific variation of the configuration for one of the tests. The required logic could be explicit (as proposed) or implicit (defined by a flag as you recommend). A single env var should have enough context to explain its usage which in this case may complicate its definition. 

For example, the name of the single env var could be `WINDOWS_SERVER_2019_ENABLED` which is tightened to a specific version of Windows Server (2019). The latter, cannot be extended to other versions, where we may end up with multiple env vars to support multiple versions, followed by several conditional statements in the WMCO's e2e code.

To change the "default" container image being used, no release PR is needed. To adjust the corner case, yes. Both cases, a PR is required. 

The [introduction](https://github.com/openshift/windows-machine-config-operator/pull/669/files#diff-43febcc9af1775a8b8390c4d64b28993e558e2e58476773d1e1d4c78b5ff0b75R49) of the `VM_TEMPLATE` env var for vSphere, was the first stepping stone towards moving the Windows Server image definition to the release repo for the e2e test. I feel the proposed env vars align with that approach.

Reply:
I would be more okay with this if it was just describing infrastructure, like the vSphere tests you point to do. But its not just infrastructure, you're moving part of the test workload definition to the release job as well with `WINDOWS_CONTAINER_IMAGE`. What happens if we need to include more changes to the tests which are different for 2019/2022, such as using [runtimeclasses](https://docs.openshift.com/container-platform/4.10/windows_containers/scheduling-windows-workloads.html#creating-runtimeclass_scheduling-windows-workloads)?

I want to avoid more logic being moved to the release repo. So I see two possible options here, either use a simple `WINDOWS_SERVER_2019_ENABLED`, or use the Windows node status to figure out the Windows server version. The first option is a lot simpler.

Reply:
Makes sense, let me adjust that.